rerun_js/web-viewer/bundle.mjs

```diff
@@ -6,25 +6,31 @@
 import { fileURLToPath } from "node:url";
 import * as path from "node:path";
 import * as fs from "node:fs";
+import * as zlib from "node:zlib";
 import * as util from "node:util";
 
 const __filename = path.resolve(fileURLToPath(import.meta.url));
 const __dirname = path.dirname(__filename);
 
-const wasm = fs.readFileSync(path.join(__dirname, "re_viewer_bg.wasm"));
+const wasm = zlib.gzipSync(fs.readFileSync(path.join(__dirname, "re_viewer_bg.wasm")));
 const js = fs.readFileSync(path.join(__dirname, "re_viewer.js"), "utf-8");
 const index = fs.readFileSync(path.join(__dirname, "index.js"), "utf-8");
 
 const INLINE_MARKER = "/*<INLINE-MARKER>*/";
 
 /** @param {Buffer} buffer */
 function buffer_to_data_url(buffer) {
-  return `data:application/wasm;base64,${buffer.toString("base64")}`;
+  return `data:application/octet-stream;gzip;base64,${buffer.toString("base64")}`;
 }
 
-async function data_url_to_buffer(dataUrl) {
-  const response = await fetch(dataUrl);
-  return response.arrayBuffer();
+async function compressed_data_url_to_buffer(dataUrl) {
+    const response = await fetch(dataUrl);
+    const blob = await response.blob();
+
+    let ds = new DecompressionStream("gzip");
+    let decompressedStream = blob.stream().pipeThrough(ds);
+
+    return await new Response(decompressedStream).arrayBuffer();
```

Comment:
```suggestion
    return new Response(decompressedStream).arrayBuffer();
```

the extra `await` after `return` adds a microtick of latency.

(think of `async` as a transform that turns the code after every `await` into a `.then(context => ...)`, you're adding an extra `then` at the end)